app-modules/knowledge-base/src/Filament/Actions/DraftWithAiAction.php

```diff
@@ -0,0 +1,146 @@
+<?php
+
+/*
+<COPYRIGHT>
+
+    Copyright © 2016-2024, Canyon GBS LLC. All rights reserved.
+
+    Advising App™ is licensed under the Elastic License 2.0. For more details,
+    see https://github.com/canyongbs/advisingapp/blob/main/LICENSE.
+
+    Notice:
+
+    - You may not provide the software to third parties as a hosted or managed
+      service, where the service provides users with access to any substantial set of
+      the features or functionality of the software.
+    - You may not move, change, disable, or circumvent the license key functionality
+      in the software, and you may not remove or obscure any functionality in the
+      software that is protected by the license key.
+    - You may not alter, remove, or obscure any licensing, copyright, or other notices
+      of the licensor in the software. Any use of the licensor’s trademarks is subject
+      to applicable law.
+    - Canyon GBS LLC respects the intellectual property rights of others and expects the
+      same in return. Canyon GBS™ and Advising App™ are registered trademarks of
+      Canyon GBS LLC, and we are committed to enforcing and protecting our trademarks
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    https://www.canyongbs.com or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+namespace AdvisingApp\KnowledgeBase\Filament\Actions;
+
+use Closure;
+use Filament\Forms\Get;
+use Filament\Forms\Set;
+use Laravel\Pennant\Feature;
+use App\Settings\LicenseSettings;
+use Filament\Resources\Pages\Page;
+use Filament\Support\Enums\MaxWidth;
+use Illuminate\Support\Facades\Vite;
+use AdvisingApp\Ai\Models\AiAssistant;
+use AdvisingApp\Ai\Settings\AiSettings;
+use Filament\Forms\Components\Textarea;
+use Filament\Notifications\Notification;
+use Filament\Forms\Components\Actions\Action;
+use AdvisingApp\Authorization\Enums\LicenseType;
+use AdvisingApp\Ai\Exceptions\MessageResponseException;
+use AdvisingApp\Ai\Settings\AiIntegratedAssistantSettings;
+
+class DraftWithAiAction extends Action
```

Comment:
Let's rename this `DraftKnowledgeBaseArticleWithAiAction` and related blade file, etc. so we can differentiate it from others.